node/src/accountant/db_access_objects/sent_payable_dao.rs

```diff
@@ -978,6 +1007,43 @@ mod tests {
         assert_eq!(retrieved_txs, vec![tx1, new_tx2, new_tx3]);
     }
 
+    #[test]
+    fn replace_records_uses_single_sql_statement() {
+        let prepare_params = Arc::new(Mutex::new(vec![]));
+        let setup_conn = Connection::open_in_memory().unwrap();
+        setup_conn
+            .execute("CREATE TABLE example (id integer)", [])
+            .unwrap();
+        let stmt = setup_conn.prepare("SELECT id FROM example").unwrap();
+        let wrapped_conn = ConnectionWrapperMock::default()
+            .prepare_params(&prepare_params)
+            .prepare_result(Ok(stmt));
+        let subject = SentPayableDaoReal::new(Box::new(wrapped_conn));
+        let tx1 = TxBuilder::default().hash(make_tx_hash(1)).nonce(1).build();
+        let tx2 = TxBuilder::default().hash(make_tx_hash(2)).nonce(2).build();
+        let tx3 = TxBuilder::default().hash(make_tx_hash(3)).nonce(3).build();
+
+        let _ = subject.replace_records(&[tx1, tx2, tx3]);
+
+        let captured_params = prepare_params.lock().unwrap();
+        let sql = &captured_params[0];
+        assert_eq!(captured_params.len(), 1);
```

Comment:
This assertion should be at the end. So that you allow to check the values from the first capture before this trivial assertion blows off.